qa/pull-tester/install-deps.sh

```diff
@@ -1,10 +1,11 @@
 #!/bin/bash
 
 # installs test dependencies
-
-curl -L https://github.com/langerhans/ltc-scrypt/archive/master.tar.gz --output master.tar.gz
-echo "ade3cdf498927990b6d153d74f0da104114e838584be5a81bef8972accd03341 master.tar.gz" | sha256sum -c
-tar zxf master.tar.gz
-pushd ltc-scrypt-master
-python3 setup.py install --user
+file=v1.0.1.tar.gz
+workdir=$(git rev-parse --show-toplevel 2> /dev/null)/qa/pull-tester
+curl -L -O https://github.com/dogecoin/ltc-scrypt/archive/refs/tags/$file
+echo "e866ade37fb27439ae0ca32f1ee4ad32be428c1fdac9bcc988b36c68648ff0de  $file" | sha256sum -c
+tar zxf $file -C $workdir/ --transform 's/ltc-scrypt-1.0.1/ltc-scrypt/' && rm -rf $file
+pushd $workdir/ltc-scrypt
+python3 -m pip install .
```

Comment:
Minor undocumented change here: the difference with this installation method (`pip install .`) and the previous one (`setup.py install --user`) is that this does not explicitly install into userspace but only defaults to it if the user that runs the script does not have write access to the system install path. Is it possible to make the userspace install explicit just in case?

Reply:
Ah! Good point. I can affix with `--user` to explicitly enforce (`python3 -m pip install . --user`) but also don't want to presume folks have pip installed despite the setuptools deprecation warnings to shift to standards-based tools:
```
/mnt/source/repos/dogecoin/qa/pull-tester/ltc-scrypt /mnt/source/repos/dogecoin
running install
/home/bluezr/.local/lib/python3.10/site-packages/setuptools/command/install.py:34: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.
  warnings.warn(
/home/bluezr/.local/lib/python3.10/site-packages/setuptools/command/easy_install.py:144: EasyInstallDeprecationWarning: easy_install command is deprecated. Use build and pip and other standards-based tools.
  warnings.warn(
/home/bluezr/.local/lib/python3.10/site-packages/pkg_resources/__init__.py:123: PkgResourcesDeprecationWarning: 1.16.0-unknown is an invalid version and will not be supported in a future release
  warnings.warn(
/home/bluezr/.local/lib/python3.10/site-packages/pkg_resources/__init__.py:123: PkgResourcesDeprecationWarning: 0.1.43ubuntu1 is an invalid version and will not be supported in a future release
  warnings.warn(
/home/bluezr/.local/lib/python3.10/site-packages/pkg_resources/__init__.py:123: PkgResourcesDeprecationWarning: 1.1build1 is an invalid version and will not be supported in a future release
```

Would that work or should I revert back? Seeing tests failing anyway so perhaps I should revert.

Reply:
I'd recommend it with the `--user` flag, as that would make us compatible with python 3.14+ too ❤️

The CI seems to fail on not being able to find distutils, can be fixed by adding `python3-distutils` from apt

Reply:
I've appended the installation directive in install-deps.sh with `--user` flag and added `python3-pip` and `python3-setuptools` to the 3 that were previously failing in ci.yml and they now pass. 

I'll have to reconfirm but I also needed `python3-dev` instead of `python3` because the Python.h header was unavailable during installation when building manually in an i386/ubuntu:bionic container early last night. May just be able to install specifically for that as oppose to change for every matrix.host if that's preferred. Otherwise can wait till next iteration of ltc-scrypt to see if that will allow avoiding any amendments in ci.yml.